content/blog/10-pearls-with-azure-functions-in-pulumi/index.md

```diff
@@ -0,0 +1,361 @@
+
+- Combined code binaries (data plane) and infrastructure (control plane) as a single unit of deployment
+- Eliminate the need for boilerplate configuration like `host.json` and `function.json` files
+- Robust typing out-of-the-box: For instance, you can flawlessly "dot into" the `content` and `req` object above.
+
+You can read more about the motivation in [Serverless as Simple Callbacks with Pulumi and Azure Functions]({{< relref "serverless-as-simple-callbacks-with-pulumi-and-azure-functions" >}}).
+
+The previous example deployed a Function App with a single Function. However, Azure supports applications with multiple Functions bundled together.
+
+### 4. REST APIs as Multiple Functions
+
+It's common to use Azure Functions for implementing RESTful APIs. As an API may consist of multiple endpoints, we may need to combine several HTTP Functions into a single deployment unit.
+
+To achieve that, we define an `HttpFunction` object per Azure Function, each with its callback and settings. Then, we pass an array of these objects into a `MultiCallbackFunctionApp` constructor. Each Function definition may have a specific route and HTTP methods to handle:
+
+``` ts
+const get = new azure.appservice.HttpFunction("Read", {
+   route: "items",
+   methods: ["GET"],
+   callback: async (context, request) => {
+       const items = await repository.list();
+       return { status: 200, body: items };
+   },
+});
+
+const post = new azure.appservice.HttpFunction("Add", {
+   route: "items",
+   methods: ["POST"],
+   callback: async (context, request) => {
+       const id = await repository.add(request.body);
+       return { status: 201, body: { id }  };
+   },
+});
+
+const app = new azure.appservice.MultiCallbackFunctionApp("multi-app", {
+   resourceGroupName: resourceGroup.name,
+   functions: [get, post],
+});
+```
+
+You can combine as many Functions as you want, including functions of different types, as shown in the next example.
+
+### 5. Function "Warming" with a Timer Function
+
+[ [Runnable Example](https://github.com/pulumi/pulumi-azure/blob/master/examples/http-multi) ]
+
+Scheduled jobs are another frequent use case for serverless functions. It's possible to define a [*cron expression*](https://docs.microsoft.com/azure/azure-functions/functions-bindings-timer#cron-expressions) and get the code executed at designated intervals.
+
+The Consumption Plan disposes a worker if no Function runs in about 20 minutes. The next execution causes a [*cold start*](https://mikhail.io/serverless/coldstarts/define/), and the response latency to be high.
+
+The following example might seem unexpected, but it's a well-known pattern in the world of Azure Functions. We combine a target HTTP Function with a Timer Function executed once in several minutes. The body of the Timer Function is empty. Its sole purpose is to trigger the Function App at a regular cadence to keep the worker "warm":
```

Comment:
```suggestion
In order to avoid this disposal, we can combine a target HTTP Function with a Timer Function. The body of the Timer Function is empty, since its sole purpose is to trigger the Function App and keep the worker "warm":
```